Dockerfile

```diff
@@ -0,0 +1,35 @@
+FROM rust:1.74-alpine3.18 as builder
+
+RUN mkdir -p ~/.cargo && \
+    echo '[registries.crates-io]' > ~/.cargo/config && \
+    echo 'protocol = "sparse"' >> ~/.cargo/config
+
+RUN apk add --no-cache libc-dev
+
+RUN USER=root cargo new --bin /app
```

Comment:
Could we create and use a non-root user? That way we follow the principle of least privilege.

Reply:
Fair point